### PR TITLE
Muartem

### PR DESCRIPTION
**Pull request by muartem**
Added AuthUI

### CODE REVIEW

**Review by matroskin062 — COMMENTED** [review]
please remove .idea folder

File: src/js/AuthUI.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  static user;
+
+  static pass;
+
+  static error = document.querySelector('#auth-error');
+
+  static main = document.querySelector('#main');
+
+  static regExpPass = new RegExp(
+    /^(?=.*[a-zA-Z])(?=.*[0-9!@#$%^&*?+])(?!.*[()_-`\\/"'|[]}{:;>.<,])(?!.*\s)(?!.*\s).{8,20}$/
```
Comment (matroskin062) [inline src/js/AuthUI.js:11]: I think it's better just to check is string empty

Comment (matroskin062) [inline src/js/AuthUI.js:11]: I think it's better just to check is string empty

File: src/js/SignInUI.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import Auth from './AuthUI';
+import AuthAPI from './AuthAPI';
+
+export default class SignIn extends Auth {
+  static user = document.querySelector('#user');
+
+  static pass = document.querySelector('#pass');
+
+  static keep = document.querySelector('#keep');
```
Comment (alexgul1) [inline src/js/SignInUI.js:9]: why do we need a `keep` field?

File: src/js/SignUpUI.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  static email = document.querySelector('#email-up');
+
+  static signUp() {
+    const { user, pass, email } = SignUp;
+    if (user.value.length < 3)
+      Auth.error.innerText = `Login at least three letter.`;
+    else if (!Auth.regExpPass.test(pass.value))
+      Auth.error.innerText = `Password must contain at least one letter, at least one number, and be longer than six charaters.`;
+    else if (!Auth.regExpEmail.test(email.value))
+      Auth.error.innerText = `Incorrect email.`;
```
Comment (alexgul1) [inline src/js/SignUpUI.js:19]: I think that this is bad validation practice

File: src/js/SignInUI.js (hunk 2 above)
Comment (muartem) [reply src/js/SignInUI.js:9]: To keep user in system.

File: src/js/AuthUI.js (hunk 1 above)
Comment (muartem) [reply src/js/AuthUI.js:11]: as you wish

File: src/js/SignUpUI.js (hunk 3 above)
Comment (muartem) [reply src/js/SignUpUI.js:19]: why?

Comment (alexgul1) [reply src/js/SignUpUI.js:19]: I think validation should be a separate method

**Comment by matroskin062** [comment]
As i understand it's old branch? So we can close this PR?

**Pull request closed without merging**